File: app/src/main/java/com/cyl/musiclake/ui/music/search/SearchContract.java
package com.cyl.musiclake.ui.music.search;


import com.cyl.musiclake.base.BaseContract;
import com.cyl.musiclake.data.db.Music;
import com.cyl.musiclake.db.SearchHistoryBean;

import java.util.List;

public interface SearchContract {

    interface View extends BaseContract.BaseView {

        void showSearchResult(List<Music> list);

        void showSearchSuggestion(List<SearchHistoryBean> list);

        void showEmptyView();
    }

    interface Presenter extends BaseContract.BasePresenter<View> {

        void search(String key, SearchEngine.Filter filter, int limit, int page);

        void getSuggestions(String query);

        void saveQueryInfo(String query);

    }
}

File: app/src/main/java/com/cyl/musiclake/ui/music/search/SearchContract.kt
package com.cyl.musiclake.ui.music.search


import com.cyl.musiclake.ui.base.BaseContract
import com.cyl.musiclake.bean.HotSearchBean
import com.music.lake.musiclib.bean.BaseMusicInfo
import com.cyl.musiclake.bean.SearchHistoryBean

interface SearchContract {

    interface View : BaseContract.BaseView {

        fun showSearchResult(list: MutableList<BaseMusicInfo>)
        fun showHotSearchInfo(list: MutableList<HotSearchBean>)
        fun showSearchHistory(list: MutableList<SearchHistoryBean>)
    }

    interface Presenter : BaseContract.BasePresenter<View> {

        fun search(key: String, filter: SearchEngine.Filter, limit: Int, page: Int)

        fun searchByType(key: String, offset: Int, type: Int)

        fun searchLocal(key: String)

        fun getHotSearchInfo()

        fun saveQueryInfo(query: String)

        fun getSearchHistory()

    }
}
